src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -45,7 +45,7 @@ internal class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
         private readonly IEnvironment _environment;
         private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
         private readonly ISharedMemoryManager _sharedMemoryManager;
-        private readonly List<TimeSpan> _workerStatusLatencyHistory = new List<TimeSpan>();
+        private readonly ConcurrentQueue<TimeSpan> _workerStatusLatencyHistory = new ConcurrentQueue<TimeSpan>();
```

Comment:
There's already a synchronization approach being used for this List - _syncLock. You just need to put a lock in GetLatencies as well, and in the lock to a ToArray() on the list to get the results to return.

Reply:
Good point.